internal/bminventory/inventory.go

```diff
@@ -5113,15 +5133,33 @@ func (b *bareMetalInventory) UpdateInfraEnvInternal(ctx context.Context, params
 		params.InfraEnvUpdateParams.PullSecret = pullSecretBackup
 	}
 
-	if params, err = b.validateAndUpdateInfraEnvParams(ctx, &params); err != nil {
-		return nil, common.NewApiError(http.StatusBadRequest, err)
-	}
-
 	err = b.db.Transaction(func(tx *gorm.DB) error {
 		if infraEnv, err = common.GetInfraEnvFromDB(tx, params.InfraEnvID); err != nil {
 			log.WithError(err).Errorf("failed to get infraEnv: %s", params.InfraEnvID)
 			return common.NewApiError(http.StatusNotFound, err)
 		}
+
+		var cluster *common.Cluster
+		clusterId := infraEnv.ClusterID
+		if clusterId != "" {
+			cluster, err = b.GetClusterInternal(ctx, installer.V2GetClusterParams{ClusterID: clusterId})
+			if err != nil {
+				// We don't want to fail here if cluster is not found. It's not responsability of this place
```

Comment:
But maybe we should log a warning anyway? Something that mentions the cluster not being found when it was expected, this could help with debugging.


I am aware the problem might be caught elsewhere too but anything that can help debugging?

Reply:
This part of the code was already there before, but I agree that a debug message here can help. Will add a log message in, thanks!